src/server/queueing/workers/create-dimail-mailbox.ts

```diff
@@ -0,0 +1,89 @@
+import PgBoss from "pg-boss";
+
+import { addEvent } from "@/lib/events";
+import { db } from "@/lib/kysely";
+import { EventCode } from "@/models/actionEvent";
+import { CreateDimailAdressDataSchemaType } from "@/models/jobs/services";
+import { ACCOUNT_SERVICE_STATUS, SERVICES } from "@/models/services";
+import { createMailbox } from "@lib/dimail/client";
+import { getUserBasicInfo } from "@/lib/kysely/queries/users";
+import { sendEmail } from "@/server/config/email.config";
+import { EMAIL_TYPES } from "@/server/modules/email";
+import { memberBaseInfoToModel } from "@/models/mapper";
+
+export const createDimailMailboxTopic = "create-dimail-mailbox";
+
+export async function createDimailMailbox(
+  job: PgBoss.Job<CreateDimailAdressDataSchemaType>,
+) {
+  console.log(
+    `Create dimail mailboxe for ${job.data.userUuid}`,
+    job.id,
+    job.name,
+  );
+  // get member infos
+  
+  const dbUser = await getUserBasicInfo({ uuid: job.data.userUuid });
+  if (!dbUser) {
+    throw new Error(`User ${job.data.userUuid} not found`);
+  }
+  const baseInfoUser = memberBaseInfoToModel(dbUser);
+ 
+  // determine the domain based on the user's legal status
+  const domain = baseInfoUser.legal_status.includes("public") ? "beta.gouv.fr" : "ext.beta.gouv.fr";
+  const mailboxInfos = await createMailbox({
+    user_name: baseInfoUser.username,
+    domain,
+  });
+
+  // envoi email invitation avec password
+  await sendEmail({
+      toEmail: [baseInfoUser.secondary_email],
+      type: EMAIL_TYPES.EMAIL_CREATED_DIMAIL,
+      variables: {
+        email: mailboxInfos.email,
+        password: mailboxInfos.password,
+        webmailUrl: process.env.DIMAIL_WEBMAIL_URL || "https://webmail.beta.gouv.fr/",
+      },
+    });
+  // MAJ infos base espace-membre (primary_email et primary_email_status)
+  await db
+    .updateTable("users")
+    .set({
+      primary_email: mailboxInfos.email,
+      primary_email_status: "active",
```

Comment:
```suggestion
      primary_email_status: EmailStatusCode.EMAIL_ACTIVE,
```
depuis [src/models/member.ts](https://github.com/betagouv/espace-membre-next/blob/8611d8ce98b0daf0af8a0911ae503d29280f6802/src/models/member.ts#L18)